pandas/tests/tslibs/test_conversion.py

```diff
@@ -66,3 +69,26 @@ def test_length_zero_copy(dtype, copy):
     arr = np.array([], dtype=dtype)
     result = conversion.ensure_datetime64ns(arr, copy=copy)
     assert result.base is (None if copy else arr)
+
+
+class MetaDatetime(type):
+    pass
+
+
```

Comment:
Why not FakeDatetime(datetime)?

Reply:
I was trying to more accurately match what was in the original issue, which used a metaclass, but this issue does show up in the same way on both regular inheritance and classes that use a metaclass.

I can change this to a regular class if desired, it would be testing the same behavior.